File: task 03/main.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func isPrime(num int) bool {
	if num < 2 {
		return false
	}
	for i := 2; i <= int(math.Sqrt(float64(num))); i++ {
		if num%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	var input string
	fmt.Print("Enter a number (n): ")
	fmt.Scanln(&input)

	n, err := strconv.Atoi(input)
	if err != nil || n < 2 {
		fmt.Println("Prime numbers start from 2 and above.")
		return
	}

	fmt.Printf("Prime numbers up to %d:\n", n)
	for i := 2; i <= n; i++ {
		if isPrime(i) {
			fmt.Println(i)
		}
	}
}

File: task 03/main.c
#include <stdio.h>

int isPrime(int num) {
    if (num < 2) return 0;
    for (int i = 2; i * i <= num; i++) {
        if (num % i == 0) return 0; 
    }
    return 1; 
}

int main() {
    int n;

    printf("Enter a number (n): ");
    scanf("%d", &n);

    if (n < 2) {
        printf("Prime numbers start from 2 and above.\n");
    } else {
        printf("Prime numbers up to %d:\n", n);
        for (int i = 2; i <= n; i++) {
            if (isPrime(i)) {
                printf("%d\n", i);
            }
        }
    }

    return 0;
}
